### ISSUE TITLE
Interpreter crashes when in focus and cmd-q is pressed (macOS)

### ISSUE THREAD

**Issue by dyfer** [supercollider/supercollider#3824]
Does this happen to anyone else?
Reproducer:
```
Window.new.front;
//now press Cmd-q (or Ctrl-q?)
//result:
//Interpreter has crashed or stopped forcefully. [Exit code: 11]
```
This will also happen without a gui window, as long as the interpreter (as opposed to the ide) is in focus.

Tested on recent develop, built against QT 5.9.1 on macOS. Some info from the console:
```
Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000078
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [0]
```
```
Thread 2 Crashed:
0   sclang                        	0x000000010d034e68 boost::asio::detail::kqueue_reactor::run(long, boost::asio::detail::op_queue<boost::asio::detail::scheduler_operation>&) + 40 (conditionally_enabled_mutex.hpp:53)
1   sclang                        	0x000000010d034c59 boost::asio::detail::scheduler::do_run_one(boost::asio::detail::conditionally_enabled_mutex::scoped_lock&, boost::asio::detail::scheduler_thread_info&, boost::system::error_code const&) + 313 (scheduler.ipp:36)
2   sclang                        	0x000000010d034969 boost::asio::detail::scheduler::run(boost::system::error_code&) + 201 (scheduler.ipp:154)
3   sclang                        	0x000000010cfb4739 asioFunction() + 73 (error_code.hpp:494)
4   sclang                        	0x000000010cfa2298 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)()> >(void*) + 40 (memory:2600)
5   libsystem_pthread.dylib       	0x00007fffd534993b _pthread_body + 180
6   libsystem_pthread.dylib       	0x00007fffd5349887 _pthread_start + 286
7   libsystem_pthread.dylib       	0x00007fffd534908d thread_start + 13
```

**Comment by brianlheim** [comment]
I can also reproduce this

**Comment by patrickdupuis** [comment]
I can't reproduce on Linux (Qt 5.10, Gnome 3.28).

**Comment by jamshark70** [comment]
Cmd-Q on Apple is "Quit," so, something is not being cleaned up properly when exiting this way.

Valid issue to prevent the crash, but users should close windows with Cmd-W instead.

There's no ctrl-q shortcut in sclang for Windows and Linux, so I'm quite sure it will not happen outside of MacOS. (Unless there's some Linux window manager that provides a global quit shortcut for all GUI apps -- not the case in any WM I've used.)

**Comment by adcxyz** [comment]
Still does that in 3.10.2, and actually you don't even need the window for it:

WHEN you bring the interpreter app icon to the front (with Cmd-Tab), 
and do Cmd-Q then,the interpreter dutifully quits as told. 

Proposal 1 - quick fix:
I think the Cmd-Q shortcut in the interpreter app should be disabled. 

Proposal 2 - preferred longterm  fix:
The IDE should start the interpreter invisibly, not as a separate visible "app", 
which has only a single SuperCollider Menu, with only a single shortcut - Cmd-Q...
This is a common source of confusion for beginners ...

**Comment by snappizz** [comment]
fixed #4533! :)
